tests/testsuite/artifact_dep.rs

```diff
@@ -817,8 +817,10 @@ fn lib_with_selected_dashed_bin_artifact_and_lib_true() {
                 let _b = include_bytes!(env!("CARGO_BIN_FILE_BAR_BAZ_baz-suffix"));
                 let _b = include_bytes!(env!("CARGO_STATICLIB_FILE_BAR_BAZ"));
                 let _b = include_bytes!(env!("CARGO_STATICLIB_FILE_BAR_BAZ_bar-baz"));
+                let _b = include_bytes!(env!("CARGO_STATICLIB_FILE_BAR_BAZ_bar_baz"));
                 let _b = include_bytes!(env!("CARGO_CDYLIB_FILE_BAR_BAZ"));
                 let _b = include_bytes!(env!("CARGO_CDYLIB_FILE_BAR_BAZ_bar-baz"));
+                let _b = include_bytes!(env!("CARGO_CDYLIB_FILE_BAR_BAZ_bar_baz"));
```

Comment:
I am not sure, but from the [doc of artifact dependencies](https://doc.rust-lang.org/nightly/cargo/reference/unstable.html#artifact-dependencies-environment-variables).

> Note that `<NAME>` is not modified in any way from the `name` specified in the crate supplying the artifact, or the crate name if not specified; for instance, it may be in lowercase, or contain dashes.

So this is still a breaking change and the old behavior is more straightforward to me.

Reply:
I'm not quite following this comment. Can you clarify, what is the breaking change, and which behavior is more straightforward?

FWIW, I didn't really have an answer for the question about the change in behavior. Since this is an unstable feature, we should be fine to change it, I was just tossing it out as something I noticed that would be affected. Do you have a proposal on how this should behave?

Reply:
Just feel like it introduces another inconsistency. Now we have a mix of env with uppercase and dash-replaced, but for the artifact name parts we keep original cases with dash-replaced. And the actual artifact is either dash-replaced or not depending on their crate-types.

I wish we could have no case toggle or replacement since the beginning. But yeah it is a thing we could definitely change and stick to a well-written rule.

Reply:
I am a bit lost what to do with this. I mean, yes, this entire PR changes the way how Cargo decides the default value of library targets. So yeah, it is a breaking change. However, I think we still expose compatible behavior by retaining the previous env-vars. So what exactly do you mean with "this is still a breaking change"? Can you show an example how a previously working package would no longer work with this patch?

Apart from that, with this patch applied, Cargo still uses `<NAME>` unmodified in any way from the `name` specified in the crate supplying the artifact, or the crate name if not specified.
I mean, if I understand correctly, `crate name` refers to dashes replaced with underscores in Cargo (unlike `package name`), so this PR makes the behavior match the RFC, unlike before, doesn't it?

Reply:
My apologies. I missed the compatible part. Looks like it would be sufficient. Thank you for your clarification.